extensions/data-plane/data-plane-validation-facade/data-plane-validation-server/src/main/java/org/eclipse/dataspaceconnector/dataplane/validation/server/DataPlaneValidationServerExtension.java

```diff
@@ -51,12 +51,13 @@ public String name() {
     @Override
     public void initialize(ServiceExtensionContext context) {
         var validationRules = Arrays.asList(new ContractValidationRule(contractNegotiationStore), new ExpirationDateValidationRule());
-        var tokenValidationService = new TokenValidationServiceImpl(extractPublicKey(context), validationRules);
+        var publicKey = extractPublicKey(context);
+        var tokenValidationService = new NimbusTokenValidationService(id -> publicKey, validationRules);
         var webService = context.getService(WebService.class);
         webService.registerController(new DataPlaneValidationFacadeController(tokenValidationService));
     }
 
-    private static PublicKey extractPublicKey(ServiceExtensionContext context) {
+    private static RSAPublicKey extractPublicKey(ServiceExtensionContext context) {
```

Comment:
Why don't stick with the `PublicKey` interface and avoid explicit casting? The concrete type is driven by the implementation but then we can treat it as a more generic `PublicKey`